x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/elasticsearch/CustomElandInternalServiceSettingsTests.java

```diff
@@ -0,0 +1,211 @@
+import static org.elasticsearch.xpack.inference.services.settings.InternalServiceSettings.NUM_THREADS;
+import static org.hamcrest.Matchers.is;
+
+public class CustomElandInternalServiceSettingsTests extends AbstractWireSerializingTestCase<CustomElandInternalServiceSettings> {
+
+    public static CustomElandInternalServiceSettings createRandom() {
+        var numAllocations = randomIntBetween(1, 10);
+        var numThreads = randomIntBetween(1, 10);
+        var modelId = randomAlphaOfLength(8);
+        SimilarityMeasure similarityMeasure = null;
+        Integer dims = null;
+        var isTextEmbeddingModel = randomBoolean();
+        if (isTextEmbeddingModel) {
+            similarityMeasure = SimilarityMeasure.COSINE;
+            dims = 123;
+        }
+
+        var elementType = randomFrom(DenseVectorFieldMapper.ElementType.values());
+
+        return new CustomElandInternalServiceSettings(numAllocations, numThreads, modelId, dims, similarityMeasure, elementType);
+    }
+
+    public void testFromMap_Request_CreatesSettingsCorrectly() {
+        var modelId = "model-foo";
+        var similarity = SimilarityMeasure.DOT_PRODUCT.toString();
+        var numAllocations = 1;
+        var numThreads = 1;
+        var serviceSettings = CustomElandInternalServiceSettings.fromMap(
+            new HashMap<>(
+                Map.of(
+                    ServiceFields.MODEL_ID,
+                    modelId,
+                    NUM_ALLOCATIONS,
+                    numAllocations,
+                    NUM_THREADS,
+                    numThreads,
+                    ServiceFields.SIMILARITY,
+                    similarity,
+                    ELEMENT_TYPE,
+                    DenseVectorFieldMapper.ElementType.FLOAT.toString()
+                )
+            ),
+            ConfigurationParseContext.REQUEST
+        );
+
+        assertThat(
+            serviceSettings,
+            is(
+                new CustomElandInternalServiceSettings(
+                    numAllocations,
+                    numThreads,
+                    modelId,
+                    null,
+                    SimilarityMeasure.DOT_PRODUCT,
+                    DenseVectorFieldMapper.ElementType.FLOAT
+                )
+            )
+        );
+    }
+
+    public void testFromMap_Request_DoesNotDefaultSimilarityElementType() {
+        var modelId = "model-foo";
+        var numAllocations = 1;
+        var numThreads = 1;
+        var serviceSettings = CustomElandInternalServiceSettings.fromMap(
+            new HashMap<>(Map.of(ServiceFields.MODEL_ID, modelId, NUM_ALLOCATIONS, numAllocations, NUM_THREADS, numThreads)),
+            ConfigurationParseContext.REQUEST
+        );
+
+        assertThat(serviceSettings, is(new CustomElandInternalServiceSettings(numAllocations, numThreads, modelId, null, null, null)));
+    }
+
+    public void testFromMap_Request_IgnoresDimensions() {
```

Comment:
Dave and I chatted about this offline. If `dimensions` is included in a PUT request when persisting the inference endpoint configuration, it will fail with an unknown field because of the `throwIfNotEmptyMap()` call in the service definition.